index.js

```diff
@@ -212,71 +172,64 @@
   //. `TypeClass` values may be used with [sanctuary-def][type-classes]
   //. to define parametrically polymorphic functions which verify their
   //. type-class constraints at run time.
-  function TypeClass(name, url, dependencies, test) {
-    return {
-      '@@type': 'sanctuary-type-classes/TypeClass@1',
-      'name': name,
-      'url': url,
-      'test': function(x) {
-        return dependencies.every (function(d) { return d.test (x); }) &&
-               test (x);
-      }
-    };
-  }
+  const TypeClass = (name, url, dependencies, test) => ({
+    '@@type': 'sanctuary-type-classes/TypeClass@1',
+    'name': name,
+    'url': url,
+    'test': x => dependencies.every (d => d.test (x)) && test (x),
+  });
 
   //  data Location = Constructor | Value
 
   //  Constructor :: Location
-  var Constructor = 'Constructor';
+  const Constructor = 'Constructor';
 
   //  Value :: Location
-  var Value = 'Value';
+  const Value = 'Value';
 
   //  functionName :: Function -> String
-  var functionName = has ('name', Function) ?
-    function functionName(f) { return f.name; } :
+  const functionName = (
+    has ('name', Function) ?
```

Comment:
Do we still need to consider the fallback for function names? Wasn't that an es5-support thing?

Reply:
Good point! You are correct: [ECMAScript 2015 Language Specification § 19.2.4.2][1].


[1]: https://262.ecma-international.org/6.0/#sec-function-instances-name

Reply:
#142 :thinking:

Reply:
Yeah, and depending on the outcome of this https://github.com/sanctuary-js/sanctuary-type-classes/pull/151#discussion_r618284535 discussion, we may end up removing the `functionName` function altogether.